### PR TITLE
Fix vite build breaking mechanism to provide extra resources for the editor

### PR DESCRIPTION
**Pull request by ralphwetzel**
## Description

When building a widget for Dashboard-2, `vite` places the resulting file into the `/resources` folder:

https://github.com/FlowFuse/node-red-dashboard-2-ui-example/blob/a5f443e290a4d453dac46e6ad9b9845dcc5958d9/vite.config.js#L26-L27

Unfortunately, as it's [documentation](https://vitejs.dev/config/build-options.html#build-emptyoutdir) states,
> by default, Vite will empty the outDir on build if it is inside project root.

This is unfortunate, as it deletes as well all (other) files existing in `/resources`, thus breaking the mechanism to provide additional files to [load into the editor as extra resources](https://nodered.org/docs/creating-nodes/resources).

## Change Proposal

- Rather than building directly into `/resources`, `vite` shall build into `ui/dist` directory.
- The resulting file will then be copied into `/resources`.
- The operation will be performed by `vite-plugin-static-copy`, added as `dev-dependency`.
- To [compensate for an error when trying to require an ESM only module](https://vitejs.dev/guide/troubleshooting#this-package-is-esm-only), `vite.config.js` has to be renamed to `vite.config.mjs`.
- Took the opportunity to fix as well the repo path in `package.json`, still pointing to a `flowforge` repo, non-existing.

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass **=> N/A**

### CODE REVIEW

**Comment by joepavitt** [comment]
I'm thinking the result is better served into a `/dist` folder? Rather than nested inside `/ui`? I like `/ui` being the source code only - thoughts?

**Comment by ralphwetzel** [comment]
> I like `/ui` being the source code only - thoughts?

The reason I proposed to put it into `/ui` was a similar one: All widget related stuff should stay in `/ui`, whereas the rest of the structure is "standard NR". 😉 
Surely I can agree on your view either. Additionally, I'd propose to remove `dist/*` from the `files` section in `package.json`:

https://github.com/FlowFuse/node-red-dashboard-2-ui-example/blob/a5f443e290a4d453dac46e6ad9b9845dcc5958d9/package.json#L28-L33

**Comment by joepavitt** [comment]
> All widget related stuff should stay in /ui, whereas the rest of the structure is "standard NR".

Good point, well presented. Will review and play with this tomorrow before merging, but provisionally, looks good.

File: vite.config.mjs [hunk 1]
```diff
@@ -1,16 +1,29 @@
...
 // Set to this to the name of this collection of components
 // This must match node-red-dashboard-2.widgets[libraryName] in package.json
 const LIBRARY_NAME = 'ui-example'
 
 // https://vitejs.dev/config/
 export default defineConfig({
-    plugins: [vue(), cssInjectedByJsPlugin()],
+    plugins: [
+        vue(),
+        cssInjectedByJsPlugin(),
```
Comment (joepavitt) [inline vite.config.mjs:16]: purpose of this plugin?

Comment (ralphwetzel) [reply vite.config.mjs:16]: You may tell me; this was already part of the (original) code! 😉

Comment (ralphwetzel) [reply vite.config.mjs:16]: With a bit of investigation: From [this plugin's documentation](https://www.npmjs.com/package/vite-plugin-css-injected-by-js):

> A Vite plugin that takes the CSS and adds it to the page through the JS. For those who want a single JS file.

Thus ... very useful and quite relevant for the NR use case!

**Review by cstns — CHANGES_REQUESTED** [review]
Looks good!

Please address the removal of the dist folder from the files key in package.json as you mentioned in https://github.com/FlowFuse/node-red-dashboard-2-ui-example/pull/19#issuecomment-2353406317 and we're good to go!

**Comment by ralphwetzel** [comment]
@cstns 
> removal of the dist folder
👍 

Wouldn't it make sense as well to remove `package-lock.json` from the repo?
I'd even propose to add this to `.gitignore` ?

**Comment by cstns** [comment]
Yes please! I noticed it was marked as changed in git did not do anything about it.

Please do it in two separate commits, first deletion of the .lock file then the addition in .gitignore

**Review by cstns — APPROVED** [review]
Thank you!

**Pull request merged**